ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,391 @@
+                **res_kwargs,
+            )
+            return
+
+        if not no_ria_remote and not force and ria_remote in ds_siblings:
+            yield get_status_dict(
+                status='error',
+                message="a storage-sibling '{}' is already configured. Use --force to overwrite it.".format(ria_remote),
+                **res_kwargs,
+            )
+            return
+
+        # parse target URL
+        # Note: For python API we should be able to deal with RI and its subclasses. However, currently quite a dance
+        # needed (see below)
+        src_url_ri = RI(url) if not isinstance(url, RI) else url
+        if src_url_ri.fragment:
+            # TODO: This is still somewhat fragile. If the given dataset id in fragment matches, it's actually fine.
+            #       But then: What if additional things are given (like @somebranch)?
+            #       However, ideally we expect no fragment at all.
+            lgr.warning("Ignoring unexpected URL fragment '%s'." % src_url_ri.fragment)
+
+        # check special remote config:
+        base_path = src_url_ri.path
+        #if not base_path:
+        #    # TODO: consider annexconfig the same way the special remote does (in-dataset special remote config)
+        #    base_path = ds.config.get("annex.ria-remote.{}.base-path".format(ria_sibling), None)
+        if not no_ria_remote and not ds.config.get("annex.ria-remote.{}.base-path".format(ria_remote), None):
+            yield get_status_dict(
+                status='impossible',
+                message="Missing required configuration 'annex.ria-remote.{}.base-path'".format(ria_remote),
+                **res_kwargs,
+            )
+            return
+
+        base_path = Path(base_path)
+
+        # append dataset id to url and use magic from clone-helper:
+        # TODO: This dance in URL parsing should be centralized in datalad-core
+        src_url_ri.fragment = ds.id
+        # Note: Attention, decode_source_spec changes the passed RI. That's why we read base_path before
+        # (as the RI's .path will be appended by the actual repo path afterwards):
+        target_props = decode_source_spec(src_url_ri, cfg=ds.config)
+        if target_props['type'] != 'ria':
+            raise ValueError("Not a valid RIA URL: %s. Expected: 'ria+[http|ssh|file|...]://[base-path]'")
+        # TODO: What sanity checks do we need?
+        #       - ria+ scheme really required? If not, ds.id in fragment doesn't do anything. Can still be valid? How
+        #         then to distinguish base-path from repo-path?
+
+        ssh_host = src_url_ri.hostname
+        #if not ssh_host:
+        #    ssh_host = ds.config.get("annex.ria-remote.{}.ssh-host".format(ria_sibling), None)
+        if not ssh_host:
+            lgr.info("No SSH-Host configured for {}. Assume local RIA store at {}.".format(ria_remote, base_path))
```

Comment:
If the scheme was `ria+ssh` this is not cool. This would look like `ria+ssh:///path` -- looks confusiong to me. `ria+file` seems like a clearer indicator.

Reply:
Yes. This is just catching what the special remote itself needs/relies on before invoking it. As soon as the special remote itself can handle those URLs, this should disappear altogether.

Besides, I don't see how `ria+ssh` URL would ever lead there. As far as I'm aware `ria+ssh` and `ria+file` work just as intended. See test.